### PR TITLE
Hide PII from external reviewers, along with private reviews

### LINKED ISSUE #3240: External Reviewer access to reviewer names, review outcomes, and reviews
### Description

External reviewers should not be able to view the staff member WHO has reviewed their application and the OUTCOME of reviews submitted. 

### Related
Reviewer can access any submission by directly visiting the submission detail page #2169
https://github.com/HyphaApp/hypha/pull/2170

### PR DESCRIPTION
**Pull request by wes-otf**
Fixes #3240. Adds `HIDE_IDENTITY_FROM_REVIEWERS`config option to hide the identities of staff & applicants from external reviewers. This also hides reviews that don't have visibility set to `reviewer` from the sidebar as that seemed ore intuitive.

There is also a fix here for the `Updated` field of the application where the original author will always be specified as the application editor even if someone else updated it. This is an edge case but will be helpful as we implement things like #3919.

## Test Steps

 - [ ] As an applicant, submit a new application to a fund that has external review
 - [ ] As staff, assign an external reviewer to the application & move the it into the external review phase.
 - [ ] While still staff, add a review & set the visibility to `private`
 - [ ] As the assigned external reviewer, ensure...
     - [ ] the applicant's name & other PII are not present anywhere on the submission
     - [ ] the staff review that was marked private is not visible in the `Reviews & assignees` sidebar
 - [ ] As staff, edit the previous review & set visibility to `reviewers`
 - [ ] As the reviewer, ensure the edited review is visible in the sidebar

### CODE REVIEW

**Comment by wes-otf** [comment]
rebased onto #3948 to use those features so some of the diff may be duplicated - code changes in this PR can be found [here](https://github.com/HyphaApp/hypha/pull/4018/files/e6b261ab9fe47b8c45b55cea17ce4e3a3d2da14d..0bec312fced84c9fd7cb47cc1c2ddeb59b432df7)

**Comment by wes-otf** [comment]
This is ready for review. I was going to add some unit tests for the `view_partials`, does anyone have previous examples of this they could toss my way so I could have an example for the review sidebar? @theskumar?

**Comment by frjo** [comment]
`HIDE_IDENTITY_FROM_REVIEWERS` is set to true on test now.

**Review by frjo — APPROVED** [review]

**Comment by frjo** [comment]
@wes-otf Needs a rebase.

**Comment by wes-otf** [comment]
On test in application details the `Updated by` field shows up twice but I can't recreate that locally. Will keep digging

**Comment by wes-otf** [comment]
Potentially a weird rebase/conflict into test? [Test branch](https://github.com/HyphaApp/hypha/blob/test/hypha/apply/funds/templates/funds/applicationsubmission_detail.html#L115-L122) vs [This branch](https://github.com/HyphaApp/hypha/blob/e0ff00e9ce4d78c37ebbcb40e30a1fcc32b4f56b/hypha/apply/funds/templates/funds/applicationsubmission_detail.html#L107-L113)

**Comment by frjo** [comment]
@wes-otf Looks like something I did when merging this in to test. I think I have fixed it now on test.

**Comment by frjo** [comment]
Needs a rebase.

File: hypha/apply/funds/templates/funds/includes/review_sidebar.html [hunk 1]
```diff
@@ -19,8 +19,13 @@
 
 <ul class="mt-4 reviews-sidebar" id="id_reviewers" x-data='{showHiddenReviewers: false}'>
     {% if not staff_reviewers_exist %}
-        <li class="reviews-sidebar__no-reviews">{% trans "No staff reviewers yet" %}</li>
-        <hr class="reviews-sidebar__split">
+        {% if request.user.is_org_faculty %}
+            <li class="reviews-sidebar__no-reviews">{% trans "No staff reviewers yet" %}</li>
+            <hr class="reviews-sidebar__split">
+        {% else %}
+            <li class="reviews-sidebar__no-reviews">{% trans "No public reviews yet" %}</li>
```
Comment (sandeepsajan0) [inline hypha/apply/funds/templates/funds/includes/review_sidebar.html:26]: Do we want to let them know that there might be some private reviews otherwise we could say "No reviews yet"?

**Review by sandeepsajan0 — APPROVED** [review]

File: hypha/apply/funds/templates/funds/includes/review_sidebar.html (hunk 1 above)
Comment (frjo) [reply hypha/apply/funds/templates/funds/includes/review_sidebar.html:26]: @wes-otf I think this is a good question. Should we just hide reviews completely, or should we say that there are x numbers of them but you can't view them.
